### PR TITLE
Fix unit tests for wdb_fim and wdb_parser for registry scan

### LINKED ISSUE #6343: Fim registry improvement unit tests, test_wdb_parser and test_wdb_fim file modify
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.0.0 | Syscheck | Agent | Sources | Linux |

## test_wdb_parser and test_wdb_fim

After the changes of the Fim registry improvement, it is necessary to modify some of the tests related to the following functions:
- wdb_parse_syscheck
- wdb_fim_insert_entry2
- wdb_syscheck_save2
- __wrap_sqlite3_step

### PR DESCRIPTION
**Pull request by jesusjimsa**
| Related issue  |
|---------------|
| #6343             |

## Description

This pull fixes all the unit tests for the _wdb_fim_ and _wdb_parser_ files. Additionally, some new unit tests have been added for _wdb_fim_ because of some new code in the `wdb_fim_insert_entry2` function.

Several `expect_*` functions have also been added.

This pull request closes #6343.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

### CODE REVIEW

File: src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.c [hunk 1]
```diff
@@ -158,7 +162,7 @@ cJSON* __wrap_wdbc_query_parse_json(__attribute__((unused)) int *sock,
 
     return mock_ptr_type(cJSON *);
 }
-cJSON* __wrap_wdb_exec(__attribute__((unused)) sqlite3 *db, 
+cJSON* __wrap_wdb_exec(__attribute__((unused)) sqlite3 *db,
                  const char *sql) {
```
Comment (jotacarma90) [inline src/unit_tests/wrappers/wazuh/wazuh_db/wdb_wrappers.c:166]: Parameter tab

File: src/unit_tests/wazuh_db/test_wdb_parser.c [hunk 2]
```diff
@@ -518,56 +532,56 @@ void test_integrity_clear_ok(void **state)
...
-        // cmocka_unit_test_setup_teardown(test_integrity_check_error, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_integrity_check_no_data, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_integrity_check_checksum_fail, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_integrity_check_ok, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_integrity_clear_error, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_integrity_clear_ok, test_setup, test_teardown),
-        // cmocka_unit_test_setup_teardown(test_invalid_command, test_setup, test_teardown)
+
+int main() {
+    const struct CMUnitTest tests[] = {
```
Comment (jotacarma90) [inline src/unit_tests/wazuh_db/test_wdb_parser.c:552]: Try to separate in different blocks of code for each function

Comment (jesusjimsa) [reply src/unit_tests/wazuh_db/test_wdb_parser.c:552]: All tests in _test_wdb_parser_ test the same function, `wdb_parse_syscheck` so they cannot be separated by function.

**Review by jotacarma90 — APPROVED** [review]

**Review by Molter73 — CHANGES_REQUESTED** [review]
We should add tests to cover these cases

imagen

There should be a couple of tests for the legacy `registry` entry type and test with `registry_value` should include a `value_name` field in its attributes.

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**